polyglot/piranha/src/models/unit_tests/scopes_test.rs

```diff
@@ -20,43 +20,68 @@ use {
   std::{collections::HashMap, path::PathBuf},
 };
 
-/// Positive test for the generated scope query, given scope generators, source code and position of pervious edit.
-#[test]
-fn test_get_scope_query_positive() {
-  let scope_generator_method = ScopeGenerator::new(
-    "Method",
-    vec![ScopeQueryGenerator::new(
-      "((method_declaration 
-          name : (_) @n
-                parameters : (formal_parameters
-                    (formal_parameter type:(_) @t0)
-                        (formal_parameter type:(_) @t1)
-                        (formal_parameter type:(_) @t2))) @xd2)",
-      "(((method_declaration 
-                        name : (_) @z
-                              parameters : (formal_parameters
-                                  (formal_parameter type:(_) @r0)
-                                      (formal_parameter type:(_) @r1)
-                                      (formal_parameter type:(_) @r2))) @qd)
-                  (#eq? @z \"@n\")
-                  (#eq? @r0 \"@t0\")
-                  (#eq? @r1 \"@t1\")
-                  (#eq? @r2 \"@t2\")
-                  )",
-    )],
-  );
+fn _get_class_scope() -> ScopeGenerator {
+  let scope_query_generator_class: ScopeQueryGenerator = ScopeQueryGeneratorBuilder::default()
+    .matcher("(class_declaration name:(_) @n) @c".to_string())
+    .generator(
+      "(
+    ((class_declaration name:(_) @z) @qc)
+    (#eq? @z \"@n\")
+    )"
+      .to_string(),
+    )
+    .build()
+    .unwrap();
+  ScopeGeneratorBuilder::default()
+    .name("Class".to_string())
+    .rules(vec![scope_query_generator_class])
+    .build()
+    .unwrap()
+}
 
-  let scope_generator_class = ScopeGenerator::new(
-    "Class",
-    vec![ScopeQueryGenerator::new(
-      "(class_declaration name:(_) @n) @c",
+fn _get_method_scope() -> ScopeGenerator {
+  let scope_query_generator_method: ScopeQueryGenerator = ScopeQueryGeneratorBuilder::default()
+    .matcher(
       "(
-          ((class_declaration name:(_) @z) @qc)
-          (#eq? @z \"@n\")
-          )",
-    )],
-  );
+    [(method_declaration 
+              name : (_) @n
+              parameters : (formal_parameters)@fp)
+     (constructor_declaration 
+              name: (_) @n
+              parameters : (formal_parameters)@fp)
+    ]@xdn)"
+        .to_string(),
+    )
+    .generator(
+      "(
+      [(((method_declaration 
+                name : (_) @z
+                parameters : (formal_parameters)@tp))
+        (#eq? @z \"@n\")
+        (#eq? @tp \"@fp\")                  
+        )
+       (((constructor_declaration 
+                name: (_) @z
+                parameters : (formal_parameters)@tp))
+        (#eq? @tp \"@fp\")
+        (#eq? @z \"@n\")
```

Comment:
Why is the order flipped in this case?

Reply:
no particular reason. I will address this.